### PR TITLE
Add more logging to the WaitForGenericK8sObjects meassurment

### PR DESCRIPTION
**Pull request by kisieland**
The logging added lists failed objects whenever it fails.

#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

File: clusterloader2/pkg/measurement/util/object_store_dynamic.go [hunk 1]
```diff
@@ -58,14 +58,14 @@ func NewDynamicObjectStore(ctx context.Context, dynamicClient dynamic.Interface,
 	}, nil
 }
 
-// ListConditions returns list of conditions for each object that was returned by lister.
-func (s *DynamicObjectStore) ListConditions() ([][]metav1.Condition, error) {
+// ListObjectsWithConditions returns list of objects with conditions for each object that was returned by lister.
+func (s *DynamicObjectStore) ListObjectsWithConditions() ([]ObjectSimplification, error) {
```
Comment (tosi3k) [inline clusterloader2/pkg/measurement/util/object_store_dynamic.go:62]: Nit: could we rename this method to `ListObjectSimplifications`?

I find this name more explicit and, at the same time, I'd expect that `Objects` by default have `conditions` included, hence the redundancy of `WithConditions` suffix.

File: clusterloader2/pkg/measurement/util/wait_for_conditions.go [hunk 2]
```diff
@@ -116,7 +117,7 @@ func WaitForGenericK8sObjects(ctx context.Context, dynamicClient dynamic.Interfa
 
 // countObjectsMatchingConditions counts objects that have a successful or failed condition.
 // Function assumes the conditions it looks for are mutually exclusive.
-func countObjectsMatchingConditions(objectConditions [][]metav1.Condition, successfulConditions []string, failedConditions []string) (successful int, failed int, all int) {
+func countObjectsMatchingConditions(objectsWithConditions []ObjectSimplification, successfulConditions []string, failedConditions []string) (successful []string, failed []string, all int) {
```
Comment (tosi3k) [inline clusterloader2/pkg/measurement/util/wait_for_conditions.go:120]: Nit: consider renaming `all` to `count` (and this function usage in `WaitForGenericK8sObjects`) - `all` suggests some collection while `count` (or `total`) implies it's a number.

File: clusterloader2/pkg/measurement/util/wait_for_conditions.go [hunk 3]
```diff
@@ -145,3 +146,7 @@ func countObjectsMatchingConditions(objectConditions [][]metav1.Condition, succe
 func conditionToKey(c metav1.Condition) string {
 	return fmt.Sprintf("%s=%s", c.Type, c.Status)
 }
+
+func objectToDebugName(o ObjectSimplification) string {
```
Comment (tosi3k) [inline clusterloader2/pkg/measurement/util/wait_for_conditions.go:150]: Nit: maybe rename to `String()` receiver method or `Name()` receiver method (as we do not print everything)?

File: clusterloader2/pkg/measurement/util/object_store_dynamic.go (hunk 1 above)
Comment (kisieland) [reply clusterloader2/pkg/measurement/util/object_store_dynamic.go:62]: ack, done

File: clusterloader2/pkg/measurement/util/wait_for_conditions.go (hunk 2 above)
Comment (kisieland) [reply clusterloader2/pkg/measurement/util/wait_for_conditions.go:120]: done

File: clusterloader2/pkg/measurement/util/wait_for_conditions.go (hunk 3 above)
Comment (kisieland) [reply clusterloader2/pkg/measurement/util/wait_for_conditions.go:150]: done

File: clusterloader2/pkg/measurement/util/wait_for_conditions.go [hunk 4]
```diff
@@ -116,7 +117,7 @@ func WaitForGenericK8sObjects(ctx context.Context, dynamicClient dynamic.Interfa
 
 // countObjectsMatchingConditions counts objects that have a successful or failed condition.
 // Function assumes the conditions it looks for are mutually exclusive.
-func countObjectsMatchingConditions(objectConditions [][]metav1.Condition, successfulConditions []string, failedConditions []string) (successful int, failed int, all int) {
+func countObjectsMatchingConditions(objectsWithConditions []ObjectSimplification, successfulConditions []string, failedConditions []string) (successful []string, failed []string, count int) {
```
Comment (tosi3k) [inline clusterloader2/pkg/measurement/util/wait_for_conditions.go:120]: Super-duper nit: `s/objectsWithConditions/objectSimplifications` or `s/objectsWithConditions/objects`.

Comment (kisieland) [reply clusterloader2/pkg/measurement/util/wait_for_conditions.go]: done

**Pull request merged**